interfaces/input_parameters/ring_section.py

```diff
@@ -177,98 +165,134 @@ def __init__(self, section_length, alpha_0,
 
         # Reshaping the input synchronous data to the adequate format and
         # get back the momentum program
-        if not isinstance(synchronous_data, dTypes._ring_program):
+        if not isinstance(synchronous_data,
+                          ring_programs._synchronous_data_program):
             synchronous_data \
-                = dTypes._ring_program.conversions[func_type](synchronous_data)
+                = ring_programs._synchronous_data_program._conversions[
+                    func_type](synchronous_data)
 
         self.synchronous_data = synchronous_data
 
+        # Setting orbit length
+        if orbit_bump is None:
+            self.length = ring_programs.orbit_length_program(
+                self.length_design)
+        else:
+            if not isinstance(orbit_bump, ring_programs.orbit_length_program):
+                orbit_bump = ring_programs.orbit_length_program(orbit_bump)
+            if orbit_bump.timebase == 'by_time':
+                orbit_bump[:, 1, :] += self.length_design
+            else:
+                orbit_bump += self.length_design
```

Comment:
I thought the `orbit_bump` was going to just the delta?

Reply:
It is! I am just adding the self.length_design to the user input which is a delta and setting the result as self.length. Note that I am not storing self.orbit_bump (to be discussed if we want this to be stored but now that you raise the point I think we should).

This part is one of the if statements that could be simplified mid term with a better setting of the __add__ rule of the datatypes (this is the location where if orbit_bump is passed as int, the += with a float crashes)

Reply:
Aaaah, interesting.  In that case I agree, orbit_bump should be stored and I'd aim for something like:
```python
self.length = self.length_design + self.orbit_bump
```

Reply:
Ok I stored orbit_bump as attr to RingSection, what I ended up doing is that self.length starts from a copy of orbit_bump to keep all the datatypes info which I offset with self.length_design